packages/flutter/lib/src/material/scaffold.dart

```diff
@@ -1668,9 +1801,12 @@ class ScaffoldState extends State<Scaffold> with TickerProviderStateMixin {
   /// animation), use [removeCurrentSnackBar].
   ///
   /// See [Scaffold.of] for information about how to obtain the [ScaffoldState].
-  ScaffoldFeatureController<SnackBar, SnackBarClosedReason> showSnackBar(SnackBar snackbar) {
+  ScaffoldFeatureController<SnackBar, SnackBarClosedReason> showSnackBar(SnackBar snackbar, {AnimationStatusListener listener}) {
```

Comment:
Since we are keeping this API: What happens when you add SnackBars via the new SnackbarMessenger and locally to the nearest Scaffold? Will that not cause trouble because the snackbars shown on different Scaffolds go out of sync? (e.g. one Scaffold will show the snackbar added locally, while the others already show the next snackbar in the global queue)?